### ISSUE TITLE
[RFE] Create a multi spec file generator

### ISSUE THREAD

**Issue by ingvagabund** [gofed/gofed#49]
As I golang project maintainer I would like to generate spec file not just for my project but for all dependencies that are not yet packaged in the ecosystem (e.g. Fedora). I would like to run only one command that will tell my what is missing and generate everything that is missing. If there is something that can not be automatically generated, I want to know what and how to achieve that and how to continue. So I have all spec file generated and if possible prepared for review.

This is quite challenge because I will need a dependency checker and minimizer which will tell me what is the minimal number of spec files I need to package. In real, I would like to package only those projects that are necessary, nothing else. So I will need partial spec file generators. It will get a list of packages that I need to package and generate just partial spec file. At the same time I want to have a lint that will check only the partially generated devel subpackage. At the same time I want to have a spec file regenerator that will generate additional devel subpackage (or just extend the current ones).

Another challenge will be to display all dependencies and project/spec files so I know what I have already generated (created review for) and what is still missing. Is it possible to put all relevant information into one console output and be able to continue/resume failed spec file generators of dependencies (some spec file will require manual intervention). Or choose GUI instead? Or just serialize all spec file generators and go step by step until all spec files and their corresponding rpms are generated, built and ready to review?

**Comment by fridex** [comment]
I think it would worth it to have a GUI and/or CLI wizard, which helps me (project maintainer) with packaging procedure. I would be informed about all the details what is/will be going on and asked when needed. It would be much easier and cleaner for a user in my opinion.

**Comment by ingvagabund** [comment]
I agree, it will be more transparent. However, still I would like to investigate possibilities for non-GUI solutions.

**Comment by fridex** [comment]
We can make a CLI wizard similar to `git add --interactive' or any other CLI interactive tool. Passing all options by arguments could be pretty confusing (but there could/should be such option to automatize generating spec files in the future).

**Comment by ingvagabund** [comment]
If a given project is missing commit for one of its dependencies take the date of the project's commit and find the first commit of its deps that as at least old as the date.
